src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -1,53 +1,107 @@
+import dataclasses
 import logging
 from collections.abc import Iterable
-from dataclasses import asdict
+from dataclasses import asdict, dataclass
 from datetime import datetime
 
 from databricks.sdk import WorkspaceClient
 from databricks.sdk.service.sql import ListQueryObjectsResponseQuery
+from databricks.sdk.service.workspace import Language
 
+from databricks.labs.lsql.backends import SqlBackend
 
-from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler, DirectFsAccessInQuery
+from databricks.labs.ucx.hive_metastore.table_migration_status import TableMigrationIndex
+from databricks.labs.ucx.source_code.base import CurrentSessionState
+from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler, DirectFsAccessInQuery, LineageAtom
+from databricks.labs.ucx.source_code.linters.context import LinterContext
 from databricks.labs.ucx.source_code.linters.directfs import DirectFsAccessSqlLinter
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class QueryProblem:
+    query_parent: str
+    query_name: str
+    code: str
+    message: str
+
+
 class QueryLinter:
 
     def __init__(
         self,
         ws: WorkspaceClient,
+        migration_index: TableMigrationIndex,
         directfs_crawler: DirectFsAccessCrawler,
     ):
         self._ws = ws
+        self._migration_index = migration_index
         self._directfs_crawler = directfs_crawler
 
-    def collect_dfsas_from_queries(self) -> Iterable[DirectFsAccessInQuery]:
+    def refresh_report(self, sql_backend: SqlBackend, inventory_database: str):
```

Comment:
Could you add a link about which report is refreshed?

Reply:
Does it only use the "query_problems" or also the "dfsa" table?

Reply:
both